.travis.yml

```diff
@@ -120,9 +121,15 @@ install:
     - pip install uritemplate.py==0.3.0
 
 # Run Python tests (core and addon) and JS tests
+
+after_script:
+    # This ensures failed tests are removed from the cache so they can be re-tried.
+    - inv remove_failures_from_testmon --db-path=$HOME/.cache/testmon/.testmondata_$TEST_BUILD
+
 script:
     - export COVERAGE=`if [ "$TRAVIS_BRANCH" == "master-w-coverage" ]; then echo "--coverage"; else echo ""; fi`
-    - invoke test_travis_$TEST_BUILD -n 4 $COVERAGE
+    - export TESTMON_DATAFILE=$HOME/.cache/testmon/.testmondata_$TEST_BUILD
+    - invoke test_travis_$TEST_BUILD -n 1 $COVERAGE --testmon
```

Comment:
I've dropped down to a single process here (from 4 originally)  because having multiple processes writing to a slqlite db will cause it to lock. So each branch gets it's own db file at `$HOME/.cache/testmon/.testmondata_$TEST_BUILD`. 

Really having branches isn't super necessary anymore because we are only running a fraction of all our tests, but keeping them allows us to make this PR relatively tiny.

Reply:
Yes in the sense we are not using xdist within a test branch, no in the sense that each test branch is running independently on it's own process, in it's own DB.